src/main/java/mike/TaskList.java

```diff
@@ -0,0 +1,121 @@
+package mike;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import exception.MikeCommandExecutionException;
+import task.DeadlineTask;
+import task.EventTask;
+import task.TodoTask;
+
+public class TaskList {
+    private List<task.Task> taskList;
+    private int numTasks = 0;
```

Comment:
I think its better to use numOfTask than plural form since most plural form should be used for some form of collections and not be confused between them.